recipe/meta.yaml

```diff
@@ -217,13 +223,22 @@ outputs:
         - {{ pin_subpackage('mkl-include', exact=True) }}
         - blas * mkl
     test:
+      requires:
+        - {{ compiler('c') }}
+        - cmake
+        - ninja
+        - pkg-config
+      files:
+        - tests
       commands:
         - IF NOT EXIST %PREFIX%\Library\lib\mkl*dll.lib EXIT /B 1  # [win]
         # Verify that the mkl library can be located through pkg-config
         - test -f `pkg-config --variable=libdir --dont-define-prefix mkl-sdl`/libmkl_rt${SHLIB_EXT}  # [unix]
         - for /f "usebackq tokens=*" %%a in (`pkg-config --variable=libdir --dont-define-prefix mkl-sdl`) do if not exist "%%a/mkl_rt.lib" exit 1  # [win]
-      requires:
-        - pkg-config
+        - cmake -S tests/mkl-devel -B build -G Ninja -DCMAKE_VERBOSE_MAKEFILE=ON  # [linux or win]
```

Comment:
not osx?

Reply:
Yeah, I'm not proud if this one. The generated binary is crashing and I don't know why. I didn't investigate much.

Reply:
On osx-64, I'm getting
```
+ ./build/myapp
dyld: Symbol not found: _mkl_blas_caxpy
  Referenced from: /Users/builder/jcmorin/miniconda/conda-bld/intel_repack_1699398061696/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/lib/libmkl_intel_ilp64.2.dylib
  Expected in: flat namespace
 in /Users/builder/jcmorin/miniconda/conda-bld/intel_repack_1699398061696/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/lib/libmkl_intel_ilp64.2.dylib
/Users/builder/jcmorin/miniconda/conda-bld/intel_repack_1699398061696/test_tmp/run_test.sh: line 10:  9968 Abort trap: 6           ./build/myapp
```
But I'm not too sure what's going on. It's the first time I see such an error (`Expected in: flat namespace`).

I looked online and found https://stackoverflow.com/questions/32277351/what-does-expected-in-flat-namespace-mean and https://developer.apple.com/forums/thread/723367?answerId=743270022#743270022, but I can't really decipher what this really means.

The `_mkl_blas_caxpy` symbole seems to come from `libmkl_core.dylib` which we link against, but the linker seems to drop/optimize it out.

The log:
```
[1/2] $PREFIX/bin/x86_64-apple-darwin13.4.0-clang   -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/mkl-devel-2023.1.0 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix -isysroot /opt/MacOSX10.10.sdk -mmacosx-version-min=10.9 -DMKL_ILP64 -fopenmp -MD -MT CMakeFiles/myapp.dir/app.c.o -MF CMakeFiles/myapp.dir/app.c.o.d -o CMakeFiles/myapp.dir/app.c.o -c $SRC_DIR/tests/mkl-devel/app.c
[2/2] : && $PREFIX/bin/x86_64-apple-darwin13.4.0-clang -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/mkl-devel-2023.1.0 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix -isysroot /opt/MacOSX10.10.sdk -mmacosx-version-min=10.9 -Wl,-search_paths_first -Wl,-headerpad_max_install_names -Wl,-pie -Wl,-headerpad_max_install_names -Wl,-dead_strip_dylibs -Wl,-rpath,$PREFIX/lib -L$PREFIX/lib CMakeFiles/myapp.dir/app.c.o -o myapp  $PREFIX/lib/libmkl_intel_ilp64.dylib  $PREFIX/lib/libmkl_intel_thread.dylib  $PREFIX/lib/libmkl_core.dylib  $PREFIX/lib/libiomp5.dylib  -lm  -ldl  -lpthread && :
+ otool -hV build/myapp
build/myapp:
Mach header
      magic  cputype cpusubtype  caps    filetype ncmds sizeofcmds      flags
MH_MAGIC_64   X86_64        ALL LIB64     EXECUTE    18       1744   NOUNDEFS DYLDLINK TWOLEVEL PIE
+ otool -L build/myapp
build/myapp:
	@rpath/libmkl_intel_ilp64.2.dylib (compatibility version 0.0.0, current version 0.0.0)
	@rpath/libiomp5.dylib (compatibility version 5.0.0, current version 5.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1213.0.0)
```

Reply:
Not sure that's ideal, from [here](https://stackoverflow.com/a/17232153):
> `-undefined suppress -flat_namespace`: these options have the effect of moving detection of missing symbols from build time to run time. In general you should avoid using these flags unless you have a very specific need.

Reply:
I'm just looking into this more... might be the solution...

Reply:
Note that I'm not using `-undefined suppress`

Reply:
So flat namespace means that all symbols that need to be resolved at runtime are in one big namespace with no reference to the library that should provide them. A two-level namespace records the libraries that should provide them. Since the missing symbols are expected in the "flat namespace" then AFAIU you need to tell the linker "flat-namespace" to look in that namespace.

Reply:
Anyway, 1) you're testing the runtime behaviour and it's correct and 2) you're not distributing anything that we have built ourselves here (I missed initially that this was the test binary) so I think it's all good